airbyte-bootloader/src/test/java/io/airbyte/bootloader/BootloaderAppTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.bootloader;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.airbyte.commons.version.AirbyteVersion;
+import io.airbyte.config.Configs;
+import io.airbyte.db.instance.jobs.JobsDatabaseInstance;
+import io.airbyte.db.instance.jobs.JobsDatabaseMigrator;
+import io.airbyte.scheduler.persistence.DefaultJobPersistence;
+import java.util.Optional;
+import lombok.val;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.PostgreSQLContainer;
+
+public class BootloaderAppTest {
+
+  @Test
+  void testBootloaderAppBlankDb() throws Exception {
+    val container = new PostgreSQLContainer<>("postgres:13-alpine")
+        .withDatabaseName("public")
+        .withUsername("docker")
+        .withPassword("docker");
+    container.start();
+
+    val version = "0.33.0-alpha";
+
+    val mockedConfigs = mock(Configs.class);
+    when(mockedConfigs.getConfigDatabaseUrl()).thenReturn(container.getJdbcUrl());
+    when(mockedConfigs.getConfigDatabaseUser()).thenReturn(container.getUsername());
+    when(mockedConfigs.getConfigDatabasePassword()).thenReturn(container.getPassword());
+    when(mockedConfigs.getDatabaseUrl()).thenReturn(container.getJdbcUrl());
+    when(mockedConfigs.getDatabaseUser()).thenReturn(container.getUsername());
+    when(mockedConfigs.getDatabasePassword()).thenReturn(container.getPassword());
+    when(mockedConfigs.getAirbyteVersion()).thenReturn(new AirbyteVersion(version));
+    when(mockedConfigs.runDatabaseMigrationOnStartup()).thenReturn(true);
+
+    val bootloader = new BootloaderApp(mockedConfigs);
+    bootloader.load();
+
+    val jobDatabase = new JobsDatabaseInstance(
+        container.getUsername(),
+        container.getPassword(),
+        container.getJdbcUrl()).getInitialized();
+    val jobsMigrator = new JobsDatabaseMigrator(jobDatabase, this.getClass().getName());
+    val latestJobsMigrationVersion = jobsMigrator.getLatestMigration().getVersion().getVersion();
+    assertEquals("0.29.15.001", latestJobsMigrationVersion);
+
+    // val configDatabase = new ConfigsDatabaseInstance(
```

Comment:
this is commented out now since config migrations are failing within the test as the migrations explicitly called out to env vars and I'm still figuring out how the right way to inject these.

Reply:
this might help: https://javadoc.io/static/org.junit-pioneer/junit-pioneer/0.9.0/org/junitpioneer/jupiter/SetEnvironmentVariable.html

Reply:
Neat!